[1LP][RFR] Extending test to assert FullScreen Button, Ctrl+Alt+Del Button and Connection Status, which are newly added

kedark3 opened the pull request:
Purpose or Intent
=================
__Updating tests__ WebMKS Console now supports "Connection Status Display", "CtrlAltDel Button" and "ToggleFullScreen" Button on latest 5.9 builds. Updating automation to support that.

These features are currently in 5.9 only. Soon to be added in CFME 5.8 as well.

{{ pytest: cfme/tests/infrastructure/test_webmks_console.py --use-provider vsphere65-nested }}

PRT Build : 19123
58z- had bug, and its fixed, but not present in current build, fix will be available probably in next build tomorrow and test should pass , https://bugzilla.redhat.com/show_bug.cgi?id=1500445
59z- works
upstream - inconsistent failure, it should really pass but it didn't, I just ran it locally and it passed with no issues.

otsuman commented:
@kedark3 can you check PRT failures pls?

otsuman commented on cfme/tests/infrastructure/test_webmks_console.py:
```diff
@@ -80,6 +80,9 @@ def test_webmks_vm_console(request, appliance, provider, vm_obj,
     request.addfinalizer(vm_console.close_console_window)
     request.addfinalizer(appliance.server.logout)
     try:
+        if appliance.version > '5.9':
```
use `appliance.version >= '5.9'` or `appliance.version > '5.8'` instead

quarckster replied:
It should be **REQUIRED**

kedark3 commented:
PR #6005 needs to be merged in order to get the PRT success for this PR.

jkrocil commented:
I initiated a manual retest on PRT now, after merging #6005.

jkrocil commented:
Fails on PRT.

kedark3 commented:
@jkrocil 58z is known failure and Bug was opened for it. It has been fixed, but not yet in Downstream. Looking at other failures as well.

kedark3 commented:
@jkrocil Downstream 59z passed, upstream failed, but shouldn't be a big deal, its a sporadic error.

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

mfalesni commented on cfme/tests/infrastructure/test_webmks_console.py:
```diff
@@ -97,19 +120,45 @@ def test_webmks_vm_console(request, appliance, provider, vm_obj,
+                logger.info("Output of 'who --count' is {} after login"
+                .format(result_after_login))
+                return result_before_login < result_after_login
+            except Exception as e:
```
**REQUIRED**: If you know what exception can happen here (I presume it is an `IOError`) you should use that one here so you don't catch what you don't want to catch. `Exception` catches everything.

mfalesni commented:
Elaborate the errors please.

dajoRH commented:
Lint report for commit 53e97a8c8314d169cd35654e6186d75d8a22c672:

`cfme/tests/infrastructure/test_webmks_console.py`:
- :red_circle: Line 137:33: **F841** *local variable 'e' is assigned to but never used*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
Lint report for commit 872519775a620c1c889fcda3e1a701c7000b70d7:

`cfme/tests/infrastructure/test_webmks_console.py`:
- :red_circle: Line 5:1: **N812** *lowercase imported as non lowercase*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

kedark3 commented:
@mfalesni  So far I have updated the Exception class to socketerror. Lint won't allow me to do SocketError. I renamed it as follows:
`from socket import error as socketerror`
because 'error' itself won't tell much unless someone who is reading the code goes back and check with Import section. 

Turns out SSH calls are culprit and for some reason they are failing in the PRT, didn't see it on my computer or Jenkins. Those calls cause `[104] Connection Reset by Peer` error, which I am catching and handling with `socketerror`. 

In wait_for calls, I added the `handle_exception=True` in order to handle any of those SSH socket errors.

kedark3 commented:
Downstream 58z won't pass, there is a BZ that is in POST state, fix is already in the codebase, it will appear in next build. And this error mostly occurs with fresh appliance, hence only present in PRT but not seen much in Jenkins.

kedark3 commented:
Downstream 58z won't pass, there is a BZ that is in POST state, fix is already in the codebase, it will appear in next build.

quarckster commented on cfme/tests/infrastructure/test_webmks_console.py:
```diff
@@ -55,7 +56,26 @@ def ssh_client(vm_obj, console_template):
+    server_settings.disable_server_roles('websocket')
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8', reason='Only valid for >= 5.8')
```
**OPTIONAL**
It can be removed now.

kedark3 commented:
@quarckster made required changes. kindly review.

mshriver commented:
Marking WIPTEST because of PRT results. Please include, in the main PR description, explanation of PRT results with a specific PRT build number (current is 19118) and per-stream explanation of failures.
